packages/expo-web-browser/ios/WebBrowserModule.swift

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022-present 650 Industries. All rights reserved.
+
+import SafariServices
+import ExpoModulesCore
+import AuthenticationServices
+
+final public class WebBrowserModule: Module {
+  private var currentWebBrowserSession: WebBrowserSession?
+  private var currentAuthSession: WebAuthSession?
+
+  public func definition() -> ModuleDefinition {
+    name("ExpoWebBrowser")
+
+    function("openBrowserAsync") { (url: URL, options: WebBrowserOptions, promise: Promise) throws in
+      guard self.currentWebBrowserSession?.isOpen != true else {
+        throw WebBrowserAlreadyOpenException()
+      }
+      self.currentWebBrowserSession = WebBrowserSession(url: url, options: options)
+      self.currentWebBrowserSession?.open(promise)
+    }
+    .runOnQueue(.main)
```

Comment:
Isn't that a default behaviour? 🤔

Reply:
I thought it is, but I was wrong. We use `DispatchQueue.global(qos: .default)` as a default queue, which doesn't have to run on the main thread. I think we'll need to change it in this cycle + add kind of `defaultQueue` component to set the default queue for all module's functions